PocketKit/Sources/PocketKit/MyList/ItemsList/SavedItemsList/SavedItemsListViewModel.swift

```diff
@@ -638,18 +629,25 @@ extension SavedItemsListViewModel: SavedItemsControllerDelegate {
         for type: NSFetchedResultsChangeType,
         newIndexPath: IndexPath?
     ) {
-        var snapshot = buildSnapshot()
-        let id = ItemsListCell<ItemIdentifier>.item(savedItem.objectID)
-        if snapshot.itemIdentifiers.contains(id) {
-            snapshot.reloadItems([id])
-        }
-        DispatchQueue.main.async { [weak self] in
-            self?._snapshot = snapshot
-        }
+        // no-op
     }
 
-    func controllerDidChangeContent(_ controller: SavedItemsController) {
-        itemsLoaded()
+    /**
+     Sets our custom snapshot to reload certain identifiers based on the NSFetched Results controller.
+     When reloading data in this view, always call itemController.performFetch which will end up calling this function.
+     */
+    func controller(_ controller: SavedItemsController, didChangeContentWith snapshot: NSDiffableDataSourceSnapshotReference) {
+        // Build up a snapshot for us to use
+        var newSnapshot = buildSnapshot()
```

Comment:
i've been wondering if we need to rebuild a snapshot every time, or if we can just use a copy of `_snapshot` and reload accordingly, since that _should_ be the latest snapshot.

Reply:
@dskuza I was thinking the same, but i couldnt figure out how to make a copy in Swift.

Reply:
So i was thinking the same. I couldnt find a way to copy it though.

Reply:
ok ill followup with some of that.